src/Solvers/fourier_tridiagonal_poisson_solver.jl

```diff
@@ -1,72 +1,114 @@
-using Oceananigans.Operators: Δzᵃᵃᶜ, Δzᵃᵃᶠ
+using Oceananigans.Operators: Δxᶜᵃᵃ, Δxᶠᵃᵃ, Δyᵃᶜᵃ, Δyᵃᶠᵃ, Δzᵃᵃᶜ, Δzᵃᵃᶠ
+using Oceananigans.Grids: XYRegRectilinearGrid, XZRegRectilinearGrid, YZRegRectilinearGrid, stretched_dimensions
 import Oceananigans.Architectures: architecture
 
-struct FourierTridiagonalPoissonSolver{G, B, R, S, β, T}
+struct FourierTridiagonalPoissonSolver{G, B, R, S, β, T, D}
                           grid :: G
     batched_tridiagonal_solver :: B
                    source_term :: R
                        storage :: S
                         buffer :: β
                     transforms :: T
+         tridiagonal_direction :: D
```

Comment:
```suggestion
```

we don't need this

Reply:
Thanks, good catch. It was a leftover that I forgot to change back. Fixed by https://github.com/CliMA/Oceananigans.jl/pull/3111/commits/8f6c5410fab5183f0773528d6217156c7819ce58

Reply:
@simone-silvestri it turns out that we need `tridiagonal_direction` in the module `Models`. While technically we could get rid of `tridiagonal_direction` and just call `stretched_direction(solver.grid)` everywhere that's needed, I think this makes the code more complicated.

So if it's okay with you, I think it's best to keep that property since it makes the code simpler and easier to understand. Especially since the name "`TridiagonalPoissonSolver`" implies there's a "`tridiagonal_direction`" associated with it. So I changed this back in https://github.com/CliMA/Oceananigans.jl/pull/3111/commits/9507a1640656865845b7a27d3c2f5384a6a608d4. Let me know if you disagree.

Reply:
is that not a property of the `batched_tridiagonal_solver` (i.e. `solver.batched_tridiagonal_solver.tridiagonal_direction`)? 
We could define
```
tridiagonal_direction(solver::TridiagonalPoissonSolver) = solver.batched_tridiagonal_solver.tridiagonal_direction
```
and pass it as a function. 
I would rather that than duplicating the information, but maybe it's just personal preference

Reply:
> is that not a property of the `batched_tridiagonal_solver` (i.e. `solver.batched_tridiagonal_solver.tridiagonal_direction`)? We could define
>
> ```
> tridiagonal_direction(solver::TridiagonalPoissonSolver) = solver.batched_tridiagonal_solver.tridiagonal_direction
> ```
> 
> and pass it as a function. I would rather that than duplicating the information, but maybe it's just personal preference

I guess it does come down to a personal preference. I'd rather have the information there in `FourierTridiagonalPoissonSolver`, than defining extra functions (however simple they may be). But it is a good point that that property can be accessed from `solver.batched_tridiagonal_solver`.

https://github.com/CliMA/Oceananigans.jl/pull/3111/commits/215d73160dabf7cff2f520be7352dd3c6e23207b removed `tridiagona_direciton` from the fourier solver, like you suggested, and accesses it via `solver.batched_tridiagonal_solver` when necessary. I think the code is indeed better that way :+1: